lib/Parse/ParseExpr.cpp

```diff
@@ -2841,7 +2841,10 @@ ExprResult Parser::ParseBoundsExpressionOrInteropType(const Declarator &D,
 
   TempTok = Tok;
   if (StartsRelativeBoundsClause(Tok))
-    ParseRelativeBoundsClauseForDecl(Result);
+    if (ParseRelativeBoundsClauseForDecl(Result)) {
+      Diag(Tok, diag::err_invalid_relative_bounds_clause);
```

Comment:
You don't need to issue an error message here.  I believe ParseRelativeBoundsClauseForDecl should have issued an error message if necessary.  This will just confuse the programmer who sees the message.  It is sufficient to just return an ExprError().